test/expect-tests/fiber/fiber_tests.ml

```diff
@@ -72,6 +72,51 @@ let test ?(expect_never = false) to_dyn f =
     print_endline "[FAIL] expected Never to be raised but it wasn't"
   | true, false -> print_endline "[FAIL] unexpected Never raised"
 
+let%expect_test "execution context of ivars" =
+  (* The point of this test it show that the execution context is restored when
+     a fiber that's blocked on an ivar is resumed. This means that fiber local
+     variables are visible for exmaple*)
+  let open Fiber.O in
+  let ivar = Fiber.Ivar.create () in
+  let var = Fiber.Var.create () in
+  let run_when_filled () =
+    Fiber.Var.set var 42 (fun () ->
+        let* peek = Fiber.Ivar.peek ivar in
+        assert (peek = None);
+        let* () = long_running_fiber () in
```

Comment:
I understand why this is there, but I feel like it add noise to the test. What about removing it, and adding another test checking that the variable is preserved across a `Scheduler.yield`?

Reply:
To me that seems even more indirect and relies on `yield` being implemented with ivars. By the way, the `long_running_fiber` is just insurance and isn't really necessary. We only really care that `read` is called before `fill`.